### PR TITLE
Follow up to comments on PR1559

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
cc @tgolen I'm not really sure why I didn't put these in an actions file. Maybe subconsciously was not sure where exactly they even belonged... I don't mind that action files are the only things that should set Onyx data. But still think it's often not clear how to group stuff.

### Fixed Issues
Addresses Slack Comment [here](https://expensify.slack.com/archives/C03TQ48KC/p1615589242200900?thread_ts=1615495494.156000&cid=C03TQ48KC) and @tgolen post merge review on https://github.com/Expensify/Expensify.cash/pull/1559

### Tests

1. Test various aspects of site/app navigation and verify things are working on after the changes

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

**Review by tgolen — DISMISSED** [review]

**Review by Beamanator — DISMISSED** [review]

**Comment by marcaaron** [comment]
Just a heads up I'm going to add a few more changes here based on @tgolen's comments. Small improvements and things that we didn't catch in the first pass of https://github.com/Expensify/Expensify.cash/pull/1559

**Comment by marcaaron** [comment]
Sorry thought I had taken this out of draft but I must have forgot. 

Updated!

File: src/libs/Navigation/linkingConfig.js
Comment (tgolen) [reply src/libs/Navigation/linkingConfig.js:12]: My concern with this is that there is route information here that is not contained within `ROUTES.js` so it's not apparent when `/:reportID` should be used and when it shouldn't. It looks to me like the only time that `ROUTE.REPORT` is referenced is when checking that path has `r` in the array (I always see it with `includes()`) so I wonder if there is a better way to do this to still maintain the functionality that is needed.

One solution that comes to mind is having:
```
// ROUTES.js
{
    REPORT: 'r',
    REPORT_WITH_ID: `r/:reportID`,
}
```

which would allow all information about the route to be contained in that file. Thoughts?

File: src/components/ScreenWrapper.js
Comment (tgolen) [inline src/components/ScreenWrapper.js:13]: I love how this solution turned out. It's much better than the dual components that I suggested.

**Review by tgolen — CHANGES_REQUESTED** [review]

File: src/ROUTES.js
Comment (marcaaron) [reply src/ROUTES.js:8]: Only problem with that is that `ROUTES.SETTINGS` is already `'settings'` so what convention should we use then? Perhaps something like:

```js
SETTINGS: {
    ROOT: 'settings',
},
```

I'm kind of partial to the underscore convention.

File: src/libs/Navigation/AppNavigator/createCustomModalStackNavigator.js
Comment (marcaaron) [reply src/libs/Navigation/AppNavigator/createCustomModalStackNavigator.js:59]: Sure, this is an easy enough change to make and enforces the preference to use function components.

File: src/libs/Navigation/linkTo.js
Comment (marcaaron) [reply src/libs/Navigation/linkTo.js:8]: There is one used in config.js I think, but it adds them to the end not the the front

Comment (marcaaron) [reply src/libs/Navigation/linkTo.js:8]: https://github.com/Expensify/Expensify.cash/blob/9b80316418e057de767ffd7249355d151ea9478c/src/CONFIG.js#L6-L16

File: src/libs/Navigation/linkingConfig.js
Comment (marcaaron) [reply src/libs/Navigation/linkingConfig.js:12]: To answer the first question.... yes! The `/:reportID` does work with `react-navigation` there's some more information here:

https://reactnavigation.org/docs/configuring-links/#passing-params

As for the second comment, you raise a good point and I think it does make sense to document these paths as constants with the parameters we expect to use with them. We might not run into too many cases where we are using `ROUTES.REPORT_WITH_ID`, but it seems like a good way to document how the routes will be used for now.

**Comment by marcaaron** [comment]
Thanks for the comments @tgolen ! Updated.

File: src/ROUTES.js
Comment (tgolen) [reply src/ROUTES.js:8]: Cool, works for me!

File: src/libs/Navigation/AppNavigator/createCustomModalStackNavigator.js
Comment (tgolen) [reply src/libs/Navigation/AppNavigator/createCustomModalStackNavigator.js]: Thanks!

File: src/libs/Navigation/linkTo.js
Comment (tgolen) [reply src/libs/Navigation/linkTo.js:8]: Ah yes! That's the one. It might be nice to add these to `Str` some day.

File: src/libs/Navigation/linkingConfig.js
Comment (tgolen) [reply src/libs/Navigation/linkingConfig.js]: Cool, thanks for linking that!

**Review by tgolen — APPROVED** [review]
looks awesome!

**Comment by tgolen** [comment]
@Beamanator do you want to go through this and take a quick look? Once you approve it, you can go ahead and merge it.

**Comment by Beamanator** [comment]
@tgolen yeah i'd love to check it out this morning! i'll merge when i'm done :D

**Comment by Beamanator** [comment]
Nice updates @marcaaron ! Especially love the change to `ScreenWrapper`, enabling lots of yummy refactoring :D

**Review by Beamanator — APPROVED** [review]

**Pull request merged**
